pyigm/field/igmfield.py

```diff
@@ -0,0 +1,262 @@
+        rho = self.calc_rhoimpact(gdz_gal, los_coord)  # Could add this to Table
+        #
+        gd_rho = np.where(rho < R)[0]
+        if len(gd_rho) == 0:
+            return None
+        # Return
+        return gdz_gal[gd_rho], rho[gd_rho]
+
+    def get_observed(self, theta=None, subtab=None):
+        """Generate a Table of observed targets
+
+        Optionally to an angular distance from field center
+
+        Parameters
+        ----------
+        theta : Quantity or Angle, optional
+          Angular radius
+        subtab : Table, optional
+          User may input a table for processing
+          theta is ignored
+
+        Returns
+        -------
+        obs_targ : Table
+          Sub-table of targets that have been observed within this radius
+        obs_dates : List
+          List of observing dates [eventually might add to Table]
+        indices : array
+          Indices from the target table
+        """
+        if (self.targets is None) or (self.observing is None):
+            raise ValueError('IgmGalaxyField: Need to fill the target and/or observing table first!')
+        if subtab is None:
+            if theta is None:
+                subtab = self.targets
+            else:
+                # Trim on angular cut first
+                targ_coord = SkyCoord(ra=self.targets['TARG_RA']*u.deg,
+                    dec=self.targets['TARG_DEC']*u.deg)
+                sep = self.coord.separation(targ_coord)
+                gdsep = np.where(sep < theta)[0]
+                if len(gdsep) == 0:
+                    return None
+                subtab = self.targets[gdsep]
+        else:
+            gdsep = np.arange(len(subtab)) # For indexing below
+        # Generate mask (set all to False; True is masked in numpy)
+        tmsk = np.array([False]*len(subtab))
+        # Grab those with a MASK_NAME
+        have_mask = np.where(~subtab['MASK_NAME'].mask)[0]
+        if len(have_mask) == 0:
+            warnings.warn("No sources with a MASK_NAME")
+            pdb.set_trace()
+            return None
+        # Get unique mask values
+        all_masks = subtab['MASK_NAME'][have_mask]
+        uni_masks = np.unique(all_masks)
+        obs_dict = {}
+        # Loop on these
+        for mask in uni_masks:
+            obs_dates = self.get_mask_obsdate(mask)
+            if len(obs_dates) > 0:
+                mt2 = np.where(subtab['MASK_NAME'][have_mask]==mask)
+                tmsk[have_mask[mt2]] = True
+                obs_dict[mask] = obs_dates
+        # Finish
+        return subtab[tmsk], obs_dict, gdsep[tmsk]
+
+    def get_unobserved(self, theta=None):
+        """Generate a Table of unobserved targets within an angular distance
+
+        Parameters
+        ----------
+        theta : Quantity, optional
+          Angular distance
+
+        Returns
+        -------
+        unobs_targ : Table
+          Sub-table of targets that have been observed within this radius
```

Comment:
Sub-table of targets that have not been observed in the field, within `theta` (if given).